File: car-service/lihua-car/src/main/java/com/lihua/car/entity/CarUser.java
package com.lihua.car.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * <p>
 * 前台用户表
 * </p>
 *
 * @author lihua
 * @since 2023-07-01
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
public class CarUser implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 主键
     */
    private Long id;

    /**
     * 用户名
     */
    private String username;

    /**
     * 密码
     */
    private String password;

    /**
     * 昵称
     */
    private String nickname;

    /**
     * 头像
     */
    private String avatar;

    /**
     * 性别
     */
    private String gender;

    /**
     * 用户状态
     */
    private String status;

    /**
     * 逻辑删除标志
     */
    private String delFlag;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

    /**
     * 最后登录时间
     */
    private LocalDateTime lastLoginTime;

    /**
     * 邮箱
     */
    private String email;

    /**
     * 手机号码
     */
    private String phoneNumber;

    /**
     * 备注
     */
    private String remark;

    /**
     * 注册类型 0 管理员新增，1 用户自助注册
     */
    private String registerType;

}

File: vue-car-admin/src/api/system/user/type/CarUser.ts
export interface CarUser {
    /**
     * 主键
     */
    id?: string;

    /**
     * 用户名
     */
    username?: string;

    /**
     * 密码
     */
    password?: string;

    /**
     * 昵称
     */
    nickname?: string;

    /**
     * 头像
     */
    avatar?: string;

    /**
     * 性别
     */
    gender?: string;

    /**
     * 用户状态
     */
    status?: string;

    /**
     * 邮箱
     */
    email?: string;

    /**
     * 手机号码
     */
    phoneNumber?: string;

    /**
     * 备注
     */
    remark?: string;

    /**
     * 注册类型 0 管理员新增，1 用户自助注册
     */
    registerType?: string;

    /**
     * 创建时间
     */
    createTime?: string;

    /**
     * 更新时间
     */
    updateTime?: string;

    /**
     * 最后登录时间
     */
    lastLoginTime?: string;
}

export interface CarUserVO extends CarUser {
    statusIsNormal?: boolean;
    updateStatusLoading?: boolean;
}

export interface CarUserDTO extends CarUser {
    /**
     * 当前页数
     */
    pageNum?: number;

    /**
     * 每页记录数
     */
    pageSize?: number;

    /**
     * 创建时间列表，用于日期范围查询
     */
    createTimeList?: string[];

    /**
     * 密码加密请求key
     */
    passwordRequestKey?: string;
} 